src/nav/nav.spec.ts

```diff
@@ -1094,6 +1150,253 @@ describe('nav', () => {
+        done();
+      });
+
+      // 2. switching from 1 -> 2
+      links[1].click();
+      fixture.detectChanges();
+
+      // 3. links are updated synchronously
+      expectLinks(fixture, [false, true, false]);
+
+      // 4. adding 2nd content, 1st still active
+      expectContents(fixture, ['content 1', 'content 2'], 0);
+      let [first, second] = getContents(fixture);
+      expectContentState(first, ['fade'], ['show'], '1');
+      expectContentState(second, ['fade'], ['show'], '0');
+
+      // 5. switching from 2 -> 3 immediately
+      links[2].click();
+      fixture.detectChanges();
+
+      // 6. links are updated synchronously
+      expectLinks(fixture, [false, false, true]);
+
+      // 7. removing 2nd, adding 3rd content, 1st still active
+      expectContents(fixture, ['content 1', 'content 3'], 0);
+      [first, second] = getContents(fixture);
+      expectContentState(first, ['fade'], ['show'], '1');
+      expectContentState(second, ['fade'], ['show'], '0');
+    });
+
+    it(`should reverse fade in if switched to a new pane after fading in has started already`, (done) => {
+      const fixture = TestBed.createComponent(TestAnimationComponent);
+      const {onItemHiddenSpy, onItemShownSpy, onNavHiddenSpy, onNavShownSpy} = fixture.componentInstance;
+      fixture.componentInstance.reduceMotion = false;
+      fixture.detectChanges();
+
+      // TEST: 1 is active, switching 1 -> 2, switching 2 -> 3 while fading in 2, catching nav (shown)
+      let first: HTMLElement;
+      let second: HTMLElement;
+
+      // 1. checking links have good initial values (1 is active)
+      const links = getLinks(fixture);
+      expectLinks(fixture, [true, false, false]);
+      expectContents(fixture, ['content 1']);
+      expectContentState(getContent(fixture), ['fade', 'show'], [], '1');
+
+      onNavHiddenSpy.and.callFake(hiddenId => {
+        // 5. (hidden) is fired 2 times, for 1 -> 2 them for 2 -> 3
+        // we care only about the 1 -> 2
+        if (hiddenId === 1) {
+          fixture.detectChanges();
+          expect(onItemHiddenSpy).toHaveBeenCalledWith(1);
+          expectContents(fixture, ['content 2']);
+          expectContentState(getContent(fixture), ['fade', 'show'], [], '0');
+
+          // 6. switching 2 -> 3
+          links[2].click();
+          fixture.detectChanges();
+
+          // 7. links are updated synchronously
+          expectLinks(fixture, [false, false, true]);
+
+          // 8. adding 3rd content, 2nd still active (only starting fading in)
+          expectContents(fixture, ['content 2', 'content 3'], 0);
+          [first, second] = getContents(fixture);
+          expectContentState(first, ['fade'], ['show'], '0');
+          expectContentState(second, ['fade'], ['show'], '0');
+        }
+      });
+
+      onNavShownSpy.and.callFake(shownId => {
+        // 9. (shown) is fired 2 times, for 1 -> 2, then for 2 -> 3
```

Comment:
Are you sure ?

> // TEST: 1 is active, switching 1 -> 2, switching 2 -> 3 while fading in 2, catching nav (shown)

I'd say if you click on 3 when 2 is fade in, `(shown)` shouldn't be called as it fades out before being completely shown. So I'd expect to have two `(hidden)` (for 1 and 2) and only one `(shown)` (for 3) called at the end of the test.

Reply:
you're right, the comment is wrong. it is called 1 time in reality, i'll remove the `if`

Reply:
And by the way the test is coorect as well, as I'm expecting only:
```ts
expect(onItemShownSpy).toHaveBeenCalledWith(3);
```